Introduce common `messaging.consumer.group.name` and `messaging.destination.subscription.name` instead of per-system ones (#815)
open-telemetry/semantic-conventions: docs/attributes-registry/messaging.md

|---|---|---|---|
| `messaging.batch.message_count` | int | The number of messages sent, received, or processed in the scope of the batching operation. [1] | `0`; `1`; `2` |
| `messaging.client_id` | string | A unique identifier for the client that consumes or produces a message. | `client-5`; `myhost@8742@s8083jm` |

> | `messaging.consumer.group.name` | string | The name of the consumer group message or batch consumer is associated with. | `my-group`; `indexer` |

Reviewer: ```suggestion
| `messaging.consumer.group.name` | string | The name of the consumer group with which a consumer is associated. | `my-group`; `indexer` |
```
Maybe a little bit clearer, it wasn't obvious to me what the value is of mentioning both "message or batch" consumers.

<!-- semconv registry.messaging(omit_requirement_level,tag=tech-specific-eventhubs) -->
| Attribute  | Type | Description  | Examples  |
|---|---|---|---|

> | `messaging.eventhubs.consumer.group` | string | The name of the consumer group the event consumer is associated with. | `indexer` |

Author: has never been released and does not need to be deprecated

<!-- semconv registry.messaging(omit_requirement_level,tag=tech-specific-servicebus) -->
| Attribute  | Type | Description  | Examples  |
|---|---|---|---|

> | `messaging.servicebus.destination.subscription_name` | string | The name of the subscription in the topic messages are received from. | `mySubscription` |

Author: has never been released and does not need to be deprecated
